source/ed/osm2ed.cpp

```diff
@@ -704,11 +707,6 @@ void PoiHouseNumberVisitor::node_callback(uint64_t osm_id, double lon, double la
  * We read another time ways to insert housenumbers and poi
  */
 void PoiHouseNumberVisitor::way_callback(uint64_t osm_id, const CanalTP::Tags &tags, const std::vector<uint64_t> & refs) {
-    if (tags.find("addr:housenumber") == tags.end() &&
```

Comment:
On paris.osm.pbf, the new version (filtering using `get_applicable_poi_rule` that is just changed) is 2.4s longer than the old ugly filter (no judgement).